### ISSUE TITLE
[BUG] GetStepTools type issue with invoke on latest 3.15.3

### ISSUE THREAD

**Issue by bkniffler** [inngest/inngest-js#506]
**Describe the bug**
Latest inngest js has a type issue on invoke.  It'll work fine with `3.15.2`, but not with `3.15.3`.
<img width="763" alt="Bildschirmfoto 2024-03-03 um 17 07 02" src="https://github.com/inngest/inngest-js/assets/4349324/b20b92f4-4e13-4efc-b95e-19dc5402c55c">

**To Reproduce**
Check this repro: https://github.com/bkniffler/inngest-repro/tree/type-issue, or use code
```ts
import { EventSchemas, type GetStepTools, Inngest } from 'inngest';

export type InngestOptions = {
  baseUrl?: string;
  appId: string;
  nodeEnv?: string;
  env?: {
    INNGEST_EVENT_KEY: string;
  };
};

const inngest = new Inngest({
  id: 'asd',
  schemas: new EventSchemas().fromRecord<{
    foo: { data: { foo: string } };
    bar: { data: { bar: string } };
  }>(),
});
const fn = inngest.createFunction(
  {
    id: 'asd',
  },
  {
    event: 'bar',
  },
  async ({}) => {
    return { foo: 'baz' } as const;
  }
);

type Steps = GetStepTools<typeof inngest>;

export async function useStep(step: Steps) {
  const result = await step.invoke('abc', {
    function: fn,
    data: { bar: 'asd' },
  });

  if (result.foo) console.log('Good');
}
```

**Expected behavior**
No type issue on invoke response.

**Comment by jpwilliams** [comment]
#507 should be a fix.

If you're able, @bkniffler, it'd be great if you could try it out using `npm install inngest@pr-507` 🙂

**Comment by bkniffler** [comment]
Seems to fix the issue in my repro @jpwilliams! Can you ping quickly once released? Thanks!

**Comment by jpwilliams** [comment]
@bkniffler Shipped in [v3.15.4](https://github.com/inngest/inngest-js/releases/tag/inngest%403.15.4).
